### PR TITLE
Fix hangs on invalid `v-for`

### LINKED ISSUE #12112: [vue] process hangs due to babel parser error
**Prettier 2.5.1**
[Playground1](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoAmAlgG4nkWW11l0C0AZhAE4C8xEGTxka5IYTzTpqAPStGFeYtLzMuAnEYgANCAhYYNaAGdkoPL14QA7gAUrCcyjw5beAJ7n9AI154YADWcDAAyngYcAAy4nDIXG6mcH4BwaFhWIHiAObIMLwArikgcBi+cFRUldF4UDmFeDlwAGJ8GATG9cggeIUwEHogABYwGDgA6sM08KZZYHBhzjP0M549YKY+IOLJvDD2ATkdCUklAFamAB5huThwAIqFEPCnOMn6Wbx7PXTFQ1heOIYBMaFQYMNkAAOAAMnxsyQmASwPUBcD2dHi+gAjs94IdDC5eqYOFA4JVKkNeHBcTRqYcmickIl3iVkhgaPkimy7o88fFmWd9DA8L5QeDIUgAEzCgI0HC5ADCEAwTNKpgArENCskACqilwsj4gP5wACSUGqsDCYCBRgAgpawjBPPc3skAL4eoA)

**Input 1:**

```vue
<template>
  <div>
    <div v-for=" a in ">aaaaa</div>
  </div>
</template>
```

**Output 1:**

```vue
SyntaxError: Unexpected token (1:1)
> 1 |
    | ^
```

[Playground2](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoAmAlgG4nkWW11l0C0AZhAE4C8xEGTxkaUIYTzTpqAPStGFeYtLzMuAnEYgANCAhYYNaAGdkoPL14QA7gAUrCcyjw5beAJ7n9AI154YADWcDAAyngYcAAy4nDIXG6mcH4BwaFhWIHiAObIMLwArikgcBi+cFRUldF4UDmFeDlwAGJ8GATG9cggeIUwEHogABYwGDgA6sM08KZZYHBhzjP0M549YKY+IOLJvDD2ATkdCUklAFamAB5huThwAIqFEPCnOMn6Wbx7PXTFQ1heOIYBMaFQYMNkAAOAAMnxsyQmASwPUBcD2dHi+gAjs94IdDC5eqYOFA4JVKkNeHBcTRqYcmickIl3iVkhgaPkimy7o88fFmWd9DA8L5QeDIUgAEzCgI0HC5ADCEAwTNKpgArENCskACqilwsj4gP5wACSUGqsDCYCBRgAgpawjBPPc3skAL4eoA)

**Input 2:**

```vue
<template>
  <div>
    <div v-for=" a in">aaaaa</div>
  </div>
</template>
```

**Output 2:**

```
TypeError: Cannot destructure property 'left' of '(intermediate value)(...)' as it is undefined.
    at printVueFor (https://prettier.io/lib/standalone.js:40:13897)
    at mz (https://prettier.io/lib/standalone.js:40:16762)
    at Object.yz [as embed] (https://prettier.io/lib/standalone.js:40:19230)
    at Object.printSubtree (https://prettier.io/lib/standalone.js:1:220684)
    at yD (https://prettier.io/lib/standalone.js:1:222198)
    at a (https://prettier.io/lib/standalone.js:1:221787)
    at s (https://prettier.io/lib/standalone.js:1:221610)
    at https://prettier.io/lib/standalone.js:40:9647
    at https://prettier.io/lib/standalone.js:1:219866
    at iD.each (https://prettier.io/lib/standalone.js:1:219698)
```

**Expected behavior:**

**Output 2** _should be_ the expected behavior of both **Input 1** and **Input 2** since both inputs is a vue syntax error missing the RHS of a `v-for`

**The current problem is**, while the playground here successfully return **Output 1** as an error, however, **Output 1** is thrown at the `babel-parser` level. When you try to run this **Input 1** yourself in your own CLI, with the default highlighting turned _on_, the parser error will bubbles to the top and go through the highlighting process by `codeFrameColumns (from @babel/code-frame)`, with an empty `text = ""` value.

https://github.com/prettier/prettier/blob/514046b3c70e6477cb69b4d36871d0d4c8c5e415/src/main/parser.js#L103

When you step into the implementation, the code loops infinitely inside the `tokenize*` generator function.

# TL;DR
So the current behavior is, when you run **Input 1** in a CLI terminal (i.e. save the code in a file e.g. `abc.vue` and run `$ prettier abc.vue`) with highlighting turned _on_, the process hangs, and CPU is at 100%.

### LINKED ISSUE #12114: [vue] v-for incorrectly removing commas from left hand side
**Prettier 2.5.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoAmAlgG5l0C0AZhAE4C8xIAFADR5+AIwCUZCCzJgehVAHpadElAWZcBOCpD8QELDBrQAzslB527CAHcAChYSmUeHNbwBPU7uHs8YANZwMADKeBhwADI0UHDILC7GcN6+AUHBWH7RAObIMOwArkkgcBjCcFRU5RF4UFn5eFlwAGIcGASGtcggePkwEDogABYwGDgA6oM08MYZYHDBjlP0U+5dYMZeINGJ7DC2vlltcQlFAFbGAB7B2ThwAIr5EPDHOIm6Gew7XXSFA1js0RgYxoVBgg2QAA4AAzvKyJMa+LBdf5wHZ0WK6ACOj3g+30Tm6xiYMXK5QG7Dg2JoFP2DSOSHiryKiQwNFyBWZN3uONiDJOuhgeGEwNB4KQACYBb4aDhsgBhCAYenFYwAVgG+USABUhU5GW8QD84ABJKCVWDBMAAgwAQTNwRg7luL0SAF9XUA)

**Input:**

```vue
<template>
  <div v-for="(,a,b) of c"></div>
</template>

```

**Output:**

```vue
<template>
  <div v-for="(a, b) of c"></div>
</template>

```

**Expected behavior:**

the `(,a,b)` is converted to `(a, b)` silently, which is wrong in semantics. It should be left untouched. Since the vue template compiler actually accepts it and correctly ignoring the `b`, while assigning `a` the index number.

### PR DESCRIPTION
**Pull request by fisker**
## Description

Fixes #12112
Fixes #12114

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Review by duailibe — APPROVED** [review]

File: src/language-html/syntax-vue.js [hunk 1]
```diff
@@ -50,14 +55,27 @@ function parseVueFor(value) {
...
+    ) {
+      return;
+    }
   } else {
     res.alias = alias;
   }
 
+  const left = `${[res.alias, res.iterator1, res.iterator2]
+    .filter(Boolean)
+    .join(",")}`;
```
Comment (fisker) [inline src/language-html/syntax-vue.js:71]: Why this write in template original..

**Review by lkho — APPROVED** [review]

File: src/language-html/syntax-vue.js (hunk 1 above)
Comment (duailibe) [reply src/language-html/syntax-vue.js:71]: Yeah, it makes sense to remove this template

Comment (lkho) [reply src/language-html/syntax-vue.js:71]: actually i found another bug

Comment (lkho) [reply src/language-html/syntax-vue.js:71]: it should not `.filter(Boolean)` the statements.
suppose:

```html
<div id="hello-vue" class="demo">
  <div v-for="(,a,b) of 'abcd'">
    {{a}}, {{b}}
  </div>
</div>
```
should print 
```
0,
1,
2,
3,
```
but now the if i call `parseVueFor('(,a,b) of "abcd"')` it returns undefined

Comment (lkho) [reply src/language-html/syntax-vue.js:71]: Moreover, if you run this sample `v-for="(,a,b) of 'abcd'"` in prettier@2.5.1 
[link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoAmAlgG5l0C0AZhAE4C8xIAFADR5+AIwCUZCCzJgehVAHpadElAWZcBOCpD8QELDBrQAzslB527CAHcAChYSmUeHNbwBPU7uHs8YANZwMADKeBhwADI0UHDILC7GcN6+AUHBWH7RAObIMOwArkkgcBjCcFRU5RF4UFn5eFlwAGIcGASGtcggePkwEDogABYwGDgA6oM08MYZYHDBjlP0U+5dYMZeINGJ7DC2vlltcQlFAFbGAB7B2ThwAIr5EPDHOIm6Gew7XXSFA1js0RgYxoVBgg2QAA4AAzvKyJMa+LBdf5wHZ0WK6ACOj3g+30Tm6xiYMXK5QG7Dg2JoFP2DSOSHiryKiQwNFyBWZN3uONiDJOuhgeGEwNB4KQACYBb4aDhsgBhCAYenFYwAVgG+USABUhU5GW8QD84ABJKCVWDBMAAgwAQTNwRg7luL0SAF9XUA)
it removed the first comma, which is incorect

Comment (lkho) [reply src/language-html/syntax-vue.js:71]: do you guys think this worth another issue?

Comment (lkho) [reply src/language-html/syntax-vue.js:71]: I've posted this new issue here: https://github.com/prettier/prettier/issues/12114

Comment (fisker) [reply src/language-html/syntax-vue.js:71]: I've fixed this case here.

**Review by sosukesuzuki — APPROVED** [review]

File: src/language-html/syntax-vue.js (hunk 1 above)
Comment (fisker) [reply src/language-html/syntax-vue.js]: Added a test for your case https://github.com/prettier/prettier/pull/12113/commits/6bbf42bb043622b0e30b68c342dec0c5ae01acb6

**Pull request merged**